### PR TITLE
V4: Single-file implementation of form components

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

I think the js code for our form components is straightforward but after discussing with @pngwn we can make it even easier for people by having the entire implementation in the Index.svelte file. Using a single file will mean that developers only have to add/remove props in one file as opposed to two. 

For checkboxgroup, I rewrote the check that triggers the change event to not use the JSON.stringify. Thought it was weird.

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/components/video.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ class Video(Component):
         Events.play,
         Events.pause,
         Events.end,
+        Events.upload,
```
Comment (freddyaboulton) [inline gradio/components/video.py:55]: Tiny oversight on my part

**Comment by freddyaboulton** [comment]
@pngwn I refactored Checkbox and Radio into the `BaseCheckbox` and `BaseRadio` pattern. This should be good for a review.

**Review by pngwn — APPROVED** [review]
Looks great! Thanks for taking care of these changes @freddyaboulton!

**Comment by freddyaboulton** [comment]
Thank you @pngwn ! 🙏

**Pull request merged**